### ISSUE TITLE
[BUG] Unable to display vm serial console window content on Rancher managed Harvester 

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#7220]
**Describe the bug**

When access Harvester from Rancher virtualization management.
After we created a vm, wait until it have already settle down. 

And open the console serial window, there is no content display 

  ![image](https://github.com/user-attachments/assets/d76d9302-6615-491e-8994-42a869b2b44d)

  https://github.com/user-attachments/assets/fe63afad-32d9-47b2-b554-3fddb78e8d23

**To Reproduce**

1. Prepare 3 nodes Harvester v1.4.0 cluster
2. Helm install Rancher v2.10.1-alpha1
3. Import Harvester from Rancher virtualization management
4. Access Harvester from Rancher virtualization management
5. Create an OpenSUSE leap image
6. Create a virtual machine `vm2`
7. Wait until the vm is running well
8. Open the serial console window and try to login

**Expected behavior**

Should be able to display vm serial console window content on the Rancher managed Harvester

**Support bundle**

[supportbundle_50777f72-4a54-4f12-bd0f-aee3c949498a_2024-12-20T17-51-37Z.zip](https://github.com/user-attachments/files/18213831/supportbundle_50777f72-4a54-4f12-bd0f-aee3c949498a_2024-12-20T17-51-37Z.zip)

**Environment**

- Harvester ISO version: v1.3.2
- Underlying Infrastructure: 3 nodes kvm machines
- Rancher version: v2.10.1-alpha1
- Harvester ui extension: v1.0.2

**Comment by irishgordo** [comment]
I notice this w/ even just Harvester itself, not imported into Rancher -> with `Google Chrome:  Version 131.0.6778.108 (Official Build) (64-bit)`.

However when I switch to using FireFox, I do not see any issues like:

![screenshot_from_2024-12-12_12-58-02_720](https://github.com/user-attachments/assets/5bb48a89-359d-435e-bf0b-b5d60880a8c9)

@TachunLin since you're using "Brave" browser which is based off of Google Chrome, do you notice any difference when you change browsers and switch to FireFox?

As I believe this has something to possibly just do with being visible in:
- Google Chrome Browser
- Chromium Browswer
- Brave Browser
And doesn't seem to effect FireFox.

**Comment by TachunLin** [comment]
Thanks @irishgordo for the reminder :+1:  

I tried it again on both Firefox and Chrome browser on Harvester v1.4.0 managed by Rancher v2.10.1

* Open vm serial console on Firefox, it can't display the console content 
  ![image](https://github.com/user-attachments/assets/592de9f2-9cb8-4357-83a9-cd2dc2c11ad4)

* Open vm serial console on Chrome, it also can't display the console content
  ![image](https://github.com/user-attachments/assets/e44eb444-7b24-42f5-ae45-c71463a1ef8a)

**Comment by a110605** [comment]
yeah, there is rendering issue for serial console in harvester-ui-extension. While it works well on harvester/dashboard.

We have added parent route `parent: 'blank'` for serial /vnc console in harvester-ui-extension ([commit](https://github.com/harvester/harvester-ui-extension/commit/a6b32e40ad9b237a4a1be087910364ee204b37fb))

VNC console works well, but serial console not.

**Comment by a110605** [comment]
The websocket `wss://localhost:8005/apis/subresources.kubevirt.io/v1/namespaces/default/virtualmachineinstances/eed/console` is connected and able to get the character data back. But it's not able to render in xterm console.

<img width="1293" alt="Screenshot 2025-01-09 at 12 11 07 PM" src="https://github.com/user-attachments/assets/778ab880-a0f2-4517-a5e7-dfad3a7fb1fc" />


Not sure if we should follow webocket data encoding in [ContainerShell.vue](https://github.com/rancher/dashboard/blob/master/shell/components/nav/WindowManager/ContainerShell.vue) for harvester serialConsole.vue.

cc @houhoucoop , @torchiaf

**Comment by TachunLin** [comment]
Verified **fixed** on UI extension `v1.0.3-rc1` with Harvester `v1.4-d166b637-head` on Rancher `v2.10.1`
Close this issue.


### Result

<details><summary>$\color{green}{\textsf{PASS}}$ Can display VM serial console window content on the Rancher managed Harvester $~~$</summary>

 1. Can display and operate with the VM serial console window

  ![image](https://github.com/user-attachments/assets/8c22a320-129b-49f2-be25-1a2fed9749b7)


  https://github.com/user-attachments/assets/46ea7ac2-ff00-49ae-abd4-70b0ff48b882



 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Can display VM web vnc window content on the Rancher managed Harvester $~~$</summary>

 1. Can display and operate with the VM web vnc window

  ![image](https://github.com/user-attachments/assets/b51fa7bb-ba6c-4cfd-834c-c0e4c0a7040f)


  https://github.com/user-attachments/assets/954f40a0-a708-4c4b-ae55-abff4f0026af



 </details>

### Test Information
* Test Environment: Single node harvester on kvm machine
* Harvester version: `v1.4-d166b637-head`
* Rancher version: v2.10.1
* UI extension version: `v1.0.3-rc1`


### Verify Steps
<details><summary> Display and operate the VM serial console </summary>

1. Prepare 3 nodes Harvester v1.4.0 cluster
2. Helm install Rancher v2.10.1-alpha1
3. Import Harvester from Rancher virtualization management
4. Access Harvester from Rancher virtualization management
5. Create an OpenSUSE leap image
6. Create a virtual machine vm2
7. Wait until the vm is running well
8. Open the serial console window and try to login

</details>
